frontend/src/views/CalculateNutrients/CalculateNutrients.tsx

```diff
@@ -1,103 +1,350 @@
+              }}
+              variant="secondary"
+            >
+              <FontAwesomeIcon icon={faPlus} />
+              Add Other
+            </Button>
+          </ButtonGroup>
+        </div>
+        {isDialogOpen && buttonClicked === 'fertilizer' && (
+          // if fertilizer button clicked
+          <FertilizerModal
+            initialModalData={
+              rowEditIndex !== undefined
+                ? fieldList.find((v) => v.index === rowEditIndex)
+                : undefined
+            }
+            rowEditIndex={rowEditIndex}
+            setFieldList={setFieldList}
+            isOpen={isDialogOpen}
+            onCancel={handleDialogClose}
+            modalStyle={{ width: '700px' }}
           />
-        </Banner>
-      </CardHeader>
-      <ButtonWrapper position="left">
-        <Button
-          text="Add Fertilizer"
-          size="sm"
-          handleClick={handleNext}
-          aria-label="Add Fertilizer"
-          variant="primary"
-          disabled={false}
-        />
-      </ButtonWrapper>
-      {tabs.length > 0 && (
-        <TabContentDisplay
-          tabs={tabs}
-          activeTab={activeTab}
-        />
-      )}
-      <ButtonWrapper position="right">
+        )}
+        {isDialogOpen && buttonClicked === 'manure' && (
+          <ManureModal
+            initialModalData={
+              rowEditIndex !== undefined
+                ? fieldList.find((v) => v.index === rowEditIndex)
+                : undefined
+            }
+            rowEditIndex={rowEditIndex}
+            setFieldList={setFieldList}
+            isOpen={isDialogOpen}
+            onCancel={handleDialogClose}
+            modalStyle={{ width: '700px' }}
+          />
+        )}
+        {isDialogOpen && buttonClicked === 'fertigation' && (
+          <FertigationModal
+            initialModalData={
+              rowEditIndex !== undefined
+                ? fieldList.find((v) => v.index === rowEditIndex)
+                : undefined
+            }
+            rowEditIndex={rowEditIndex}
+            setFieldList={setFieldList}
+            isOpen={isDialogOpen}
+            onCancel={handleDialogClose}
+            modalStyle={{ width: '700px' }}
+          />
+        )}
+        {isDialogOpen && buttonClicked === 'other' && (
+          <OtherModal
+            initialModalData={
+              rowEditIndex !== undefined
+                ? fieldList.find((v) => v.index === rowEditIndex)
+                : undefined
+            }
+            rowEditIndex={rowEditIndex}
+            setFieldList={setFieldList}
+            isOpen={isDialogOpen}
+            onCancel={handleDialogClose}
+            modalStyle={{ width: '700px' }}
+          />
+        )}
+      </>
+      {/* tabs = the fields the user has entered */}
+      <TabsMaterial
+        activeTab={activeField}
+        tabLabel={fieldList.length > 0 ? fieldList.map((field) => field.FieldName) : ['Field 1']}
+      />
```

Comment:
I think we should put this component listing the fields on a row below the Add buttons (or shift the add buttons down to its own row?), as increasing field numbers is going to spread right, right towards the add buttons.

![image](https://github.com/user-attachments/assets/8cc9cba0-9ef6-4100-a039-e538f24bd699)